lib/internal/source_map/prepare_stack_trace.js

```diff
@@ -22,30 +22,36 @@ const {
   kNoOverride,
   overrideStackTrace,
   maybeOverridePrepareStackTrace,
-  kIsNodeError,
 } = require('internal/errors');
 const { fileURLToPath } = require('internal/url');
 const { setGetSourceMapErrorSource } = internalBinding('errors');
 
+const MainContextObjectToString = ObjectPrototype.toString;
+
 // Create a prettified stacktrace, inserting context from source maps
 // if possible.
 const prepareStackTrace = (globalThis, error, trace) => {
   // API for node internals to override error stack formatting
   // without interfering with userland code.
   // TODO(bcoe): add support for source-maps to repl.
-  if (overrideStackTrace.has(error)) {
-    const f = overrideStackTrace.get(error);
+  const fn = overrideStackTrace.get(error);
+  if (fn !== undefined) {
     overrideStackTrace.delete(error);
-    return f(error, trace);
+    return fn(error, trace);
   }
 
   const globalOverride =
     maybeOverridePrepareStackTrace(globalThis, error, trace);
   if (globalOverride !== kNoOverride) return globalOverride;
 
   let errorString;
-  if (kIsNodeError in error) {
-    errorString = `${error.name} [${error.code}]: ${error.message}`;
+  // Do not use primordials here: we intercept user code here.
+  if (typeof error.toString === 'function' &&
+      error.toString !== MainContextObjectToString) {
+    errorString = error.toString();
```

Comment:
```suggestion
    errorString = `${error}`;
```

Reply:
i think this change still needs to be made?

Reply:
This does not apply anymore.